include/fmt/color.h

```diff
@@ -630,6 +630,118 @@ inline auto format_to(OutputIt out, const text_style& ts, const S& format_str,
                     fmt::make_format_args<buffer_context<char_t<S>>>(args...));
 }
 
+FMT_BEGIN_DETAIL_NAMESPACE
+
+template <typename Arg> struct styled_arg {
+  FMT_CONSTEXPR styled_arg(Arg const& argument, text_style style)
+      : argument(argument), style(style) {}
+
+  const Arg& argument;
+  text_style style;
+};
+
+FMT_END_DETAIL_NAMESPACE
```

Comment:
Could you move the definition of `styled_arg` to one of the `detail` blocks earlier in this file instead of introducing another one?